collector/stats.go

```diff
@@ -84,9 +86,33 @@ func getPercentile(t *topic, percentile int) float64 {
 	return 0
 }
 
-func getNsqdStats(nsqdURL string) (*stats, error) {
+type statsClient struct {
+	nsqdURL    string
+	httpClient *http.Client
+}
+
+func newStatsClient(nsqdURL string, timeout time.Duration) *statsClient {
+	return &statsClient{
+		nsqdURL: nsqdURL,
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				Dial: func(network, addr string) (net.Conn, error) {
+					c, err := net.DialTimeout(network, addr, timeout)
+					if err != nil {
+						return nil, err
+					}
+					if err := c.SetDeadline(time.Now().Add(timeout)); err != nil {
```

Comment:
In worst case canceled request time could be close to timeout*2 (dial + read)
Please consider using http.Client.Timeout instead.

Useful reference about Go's server/client timeouts: https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/